src/installer/managed/Microsoft.NET.HostModel/ComHost/ClsidMap.cs

```diff
@@ -65,7 +65,7 @@ public static void Create(MetadataReader metadataReader, string clsidMapPath)
 
             using (StreamWriter writer = File.CreateText(clsidMapPath))
             {
-                writer.Write(JsonSerializer.Serialize(clsidMap, new JsonSerializerOptions { IgnoreNullValues = true }));
+                writer.Write(JsonSerializer.Serialize(clsidMap, new JsonSerializerOptions { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingDefault }));
```

Comment:
```suggestion
                writer.Write(JsonSerializer.Serialize(clsidMap, new JsonSerializerOptions { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull }));
```

to be sure this is only applied to `null` and not value type defaults as well.

Reply:
FYI - @jkoritzinsky @swaroop-sridhar @vitek-karas 

It would be great if we could somehow fix this dependency - like we talked about in https://github.com/dotnet/runtime/issues/1823. That way this one managed library wouldn't be hanging out always out of date.

Reply:
@layomia  I actually just did what the obsolete message told me :)
https://github.com/dotnet/runtime/runs/1005102491
`ComHost/ClsidMap.cs(68,93): error CS0618: (NETCORE_ENGINEERING_TELEMETRY=Build) 'JsonSerializerOptions.IgnoreNullValues' is obsolete: 'To ignore null values when serializing, set DefaultIgnoreCondition to JsonIgnoreCondition.WhenWritingDefault.'`

Do we need to update that message?

Reply:
@ericstj - ah, thanks. Yes we do need to update the obsoletion message; I missed doing this when [`JsonIgnoreCondition.WhenWritingNull` was added](https://github.com/dotnet/runtime/pull/39388). Opened https://github.com/dotnet/runtime/pull/41141.